lib/vanagon/component/dsl.rb

```diff
@@ -241,7 +241,10 @@ def install_configfile(source, target)
       # @param target [String] path to the desired symlink
       def link(source, target)
         @component.install << "#{@component.platform.install} -d '#{File.dirname(target)}'"
-        @component.install << "ln -s '#{source}' '#{target}'"
+        # Use a bash conditional to only create the link if it doesn't already point to the correct source.
+        # This allows rerunning the install step to be idempotent, rather than failing because the link
+        # already exists.
+        @component.install << "([[ `readlink '#{target}'` == '#{source}' ]] || ln -s '#{source}' '#{target}')"
```

Comment:
readlink isn't readily available on AIX. Not sure about other platforms. Would something like `ls -l #{target} | awk '{print $NF}'` work ?